source/dagger/gameplay/team_game/golem.cpp

```diff
@@ -93,66 +88,65 @@ void ancient_defenders::GolemBehaviorSystem::Run()
 			}
 			else if (golem_.currentAction == EAction::Attacking) {
 				AnimatorPlay(animation_, "ancient_defenders:golem:ATTACK_FRONT");
-				Engine::Registry().get<MageStats>(range_.target).health -= golem_.meleeDmg;
+				Engine::Registry().get<Health>(range_.target).currentHealth -= golem_.meleeDmg* Engine::DeltaTime();
 				golem_.currentAction = EAction::Moving; // Go back to moving after attacking
 			}
 		});
 
 }
 
-void ancient_defenders::GolemBehaviorSystem::OnEndOfFrame()
-{
-
-	auto view = Engine::Registry().view<MageStats>();
-
-	auto it = view.begin();
-	while (it != view.end()) {
-		auto& en = view.get<MageStats>(*it);
-		if (en.health <= 0.0f) {
-			Engine::Registry().destroy(*it);
-		}
-		it++;
-	}
-}
-
-Golem ancient_defenders::Golem::Get(Entity entity_)
+void ancient_defenders::EnemyBehaviorSystem::OnEndOfFrame()
 {
-	auto& reg = Engine::Registry();
-	auto& sprite = reg.get_or_emplace<Sprite>(entity_);
-	auto& pos = reg.get_or_emplace<Transform>(entity_);
-	auto& anim = reg.get_or_emplace<Animator>(entity_);
-	auto& gol = reg.get_or_emplace<Enemy>(entity_);
-	auto& col = reg.get_or_emplace<SimpleCollision>(entity_);
-	auto& roa = reg.get_or_emplace<RangeOfAttack>(entity_);
-
-	return Golem{ entity_, sprite, pos, anim, gol, col, roa };
 }
 
-Golem ancient_defenders::Golem::Create()
+Entity ancient_defenders::Golem::Create()
 {
 	auto& reg = Engine::Registry();
 	auto entity = reg.create();
-	auto gol = Golem::Get(entity);
+	auto& sprite = reg.emplace<Sprite>(entity);
+	auto& coordinates = reg.emplace<Transform>(entity);
+	auto& anim = reg.emplace<Animator>(entity);
+	auto& golem = reg.emplace<Enemy>(entity);
```

Comment:
golem to enemy here as well

Reply:
Actually here the name can stay golem, since this is Golem::Create function. We had in mint adding different enemies that would have their own Create functions due to the small differences in their data/stats/assets, so leaving the name unchanged here might provide better code readability later when more enemy types are in play.